src/ripple/rpc/impl/Handler.h

```diff
@@ -94,20 +94,18 @@ conditionMet(Condition condition_required, T& context)
     }
 
     if (context.app.getOPs().isAmendmentBlocked() &&
-        (condition_required & NEEDS_CURRENT_LEDGER ||
-         condition_required & NEEDS_CLOSED_LEDGER))
+        (condition_required != NO_CONDITION))
```

Comment:
This is not the same test as before. Is the change on purpose?

Reply:
yes, I agree that the conditions have changed. I made this change approximately one year ago, so I am spacing out on the details. Let me think/check and get back to you

Reply:
Based on the old values, I think the conditions are equivalent. Between `NEEDS_CURRENT_LEDGER` and `NEEDS_CLOSED_LEDGER`, having any of the three low bits set in `condition_required` would result in the expression evaluating to true, so that covers all of the `!= NO_CONDITION` conditions. So functionally, they are equivalent.

Now whether that was the _intent_ is a different question.

It turns out there are only two operations which have `NEEDS_NETWORK_CONNECTION`: `ledger_cleaner` and `tx`. I think it's fine to disallow both of them if the server is amendment blocked.